Branch-SDK/Branch-SDK/BranchLinkProperties.m

```diff
@@ -0,0 +1,29 @@
+//
+//  BranchLinkProperties.m
+//  Branch-TestBed
+//
+//  Created by Derrick Staten on 10/16/15.
+//  Copyright © 2015 Branch Metrics. All rights reserved.
+//
+
+#import "BranchLinkProperties.h"
+
+@implementation BranchLinkProperties
+
+- (NSDictionary *)controlParams {
+    if (!_controlParams) {
+        _controlParams = [[NSDictionary alloc] init];
+    }
+    return _controlParams;
+}
+
+- (void)addControlParam:(NSString *)controlParam withValue:(NSString *)value {
+    if (!controlParam || ! value) {
```

Comment:
The backend will still have to do validation. This is the standard validation we do elsewhere in the SDK. Even checking null ness, emptiness, length... all these things could be circumvented by specifying @"-" as `controlParam` or `value`. I'm going to keep this the same as validation elsewhere in the SDK for now